app/Http/Controllers/Api/Auth/RegisterController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers\Api\Auth;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Exceptions\HttpResponseException;
+use Illuminate\Http\Request;
+use App\User;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Validator;
+
+class RegisterController extends Controller
+{
+    /**
+     * Register api
+     *
+     * @return \Illuminate\Http\Response
+     */
+
+    public function register(Request $request)
+    {
+
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'required',
+                'email' => 'required|email',
+                'password' => 'required',
+            ]
+        );
+
+        if ($validator->fails()) {
+            return response()->json(['error' => $validator->errors()], 401);
```

Comment:
errors ?

```php
return response()->json(['errors' => $validator->errors()], 401);
```

Reply:
What response structure would you like ?

Reply:
```javascript
{
  'errors' : '404 error',
  'payload': { 'id': 'zrt' } 
}
```

payload and errors can be an object, array, array of objects, a string w/e